test/40-target-state.spec.ts

```diff
@@ -0,0 +1,235 @@
+import { SinonStub, stub, spy, SinonSpy } from 'sinon';
+import { Promise } from 'bluebird';
```

Comment:
Should be able to use native Promises

Reply:
It doesn't :(  I have to mock the `getAsync` method with is the promisified version of `get` from the `request` module, and it must return a Bluebird Promise.